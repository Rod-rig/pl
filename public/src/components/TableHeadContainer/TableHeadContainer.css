@import '../../css/_variables.css';

.table-head__teamName {
    width: 100%;
}

.table-head .table-head__cell:not(.table-head__teamName) {
    text-align: center;
    min-width: 90px;
}

.table-head .table-head__position {
    width: 80px;
}

.table-head__cell svg {
    position: absolute;
    top: 50%;
    right: -20px;
    margin: -8px 0 0;
}

@media (--md-breakpoint) {
    .table-head .table-head__cell {
        padding-left: 6px;
        padding-right: 6px;
    }

    .table-head .table-head__matches,
    .table-head .table-head__goals-for,
    .table-head .table-head__goals-against {
        display: none;
    }

    .table-head .table-head__cell:not(.table-head__teamName) {
        min-width: 80px;
    }
}

@media (--sm-breakpoint) {
    .table-head .table-head__w,
    .table-head .table-head__d,
    .table-head .table-head__l {
        display: none;
    }

    .table-head .table-head__cell:last-child {
        padding-right: 6px;
    }

    .table-head .table-head__cell:not(.table-head__teamName) {
        min-width: 70px;
    }
}
