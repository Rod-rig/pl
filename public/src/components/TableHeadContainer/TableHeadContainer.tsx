import {TableCell, TableHead, TableRow, TableSortLabel, Tooltip} from 'material-ui';
import * as React from 'react';

interface IHead {
  order?: any;
  sort?: string;
  sortHandle?: any;
  chars: string[];
}

const TableHeadContainer = (props: IHead) => {
  return (
    <TableHead>
      <TableRow>
        {props.chars.map(
          (name: string, index: number) => (
            <TableCell
              key={index}
              sortDirection={props.order === 'asc' ? 'desc' : 'asc'}
              padding='checkbox'
            >
              <Tooltip title='Sort' enterDelay={300}>
                <TableSortLabel
                  active={props.sort === name}
                  direction={props.order}
                  onClick={props.sortHandle}
                >
                  {name}
                </TableSortLabel>
              </Tooltip>
            </TableCell>
          ),
        )}
      </TableRow>
    </TableHead>
  );
};

export default TableHeadContainer;
