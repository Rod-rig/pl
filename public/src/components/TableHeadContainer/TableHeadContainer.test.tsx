import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TableHeadContainer from './TableHeadContainer';

const sort = ['position', 'team', 'points'];

describe('TableContainer', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<TableHeadContainer orderBy={sort}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
