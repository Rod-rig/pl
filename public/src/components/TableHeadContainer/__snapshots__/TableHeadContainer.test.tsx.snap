// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadContainer should render correctly with another props 1`] = `
<thead
  className="MuiTableHead-root-1"
>
  <tr
    className="MuiTableRow-root-2 MuiTableRow-head-3"
  >
    <th
      aria-sort="descending"
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-paddingCheckbox-13"
      scope="col"
    >
      <div
        className="MuiTooltip-root-15"
      >
        <span
          aria-describedby={undefined}
          className="MuiButtonBase-root-29 MuiTableSortLabel-root-24"
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          position
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-32 MuiTableSortLabel-icon-26 MuiTableSortLabel-iconDirectionAsc-28"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
        <div
          className="MuiTooltip-popper-16 MuiTooltip-popperClose-17"
          data-placement={undefined}
          data-x-out-of-boundaries={undefined}
          style={
            Object {
              "left": 0,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            aria-hidden={true}
            className="MuiTooltip-tooltip-18 MuiTooltip-tooltipPlacementBottom-22"
            id={undefined}
            role="tooltip"
          >
            Sort
          </div>
        </div>
      </div>
    </th>
    <th
      aria-sort="descending"
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-paddingCheckbox-13"
      scope="col"
    >
      <div
        className="MuiTooltip-root-15"
      >
        <span
          aria-describedby={undefined}
          className="MuiButtonBase-root-29 MuiTableSortLabel-root-24"
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          team
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-32 MuiTableSortLabel-icon-26 MuiTableSortLabel-iconDirectionAsc-28"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
        <div
          className="MuiTooltip-popper-16 MuiTooltip-popperClose-17"
          data-placement={undefined}
          data-x-out-of-boundaries={undefined}
          style={
            Object {
              "left": 0,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            aria-hidden={true}
            className="MuiTooltip-tooltip-18 MuiTooltip-tooltipPlacementBottom-22"
            id={undefined}
            role="tooltip"
          >
            Sort
          </div>
        </div>
      </div>
    </th>
    <th
      aria-sort="descending"
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-paddingCheckbox-13"
      scope="col"
    >
      <div
        className="MuiTooltip-root-15"
      >
        <span
          aria-describedby={undefined}
          className="MuiButtonBase-root-29 MuiTableSortLabel-root-24 MuiTableSortLabel-active-25"
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          points
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-32 MuiTableSortLabel-icon-26 MuiTableSortLabel-iconDirectionAsc-28"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
        <div
          className="MuiTooltip-popper-16 MuiTooltip-popperClose-17"
          data-placement={undefined}
          data-x-out-of-boundaries={undefined}
          style={
            Object {
              "left": 0,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            aria-hidden={true}
            className="MuiTooltip-tooltip-18 MuiTooltip-tooltipPlacementBottom-22"
            id={undefined}
            role="tooltip"
          >
            Sort
          </div>
        </div>
      </div>
    </th>
  </tr>
</thead>
`;

exports[`TableHeadContainer should render correctly with default props 1`] = `
<thead
  className="MuiTableHead-root-1"
>
  <tr
    className="MuiTableRow-root-2 MuiTableRow-head-3"
  >
    <th
      aria-sort="ascending"
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-paddingCheckbox-13"
      scope="col"
    >
      <div
        className="MuiTooltip-root-15"
      >
        <span
          aria-describedby={undefined}
          className="MuiButtonBase-root-29 MuiTableSortLabel-root-24"
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          position
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-32 MuiTableSortLabel-icon-26 MuiTableSortLabel-iconDirectionDesc-27"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
        <div
          className="MuiTooltip-popper-16 MuiTooltip-popperClose-17"
          data-placement={undefined}
          data-x-out-of-boundaries={undefined}
          style={
            Object {
              "left": 0,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            aria-hidden={true}
            className="MuiTooltip-tooltip-18 MuiTooltip-tooltipPlacementBottom-22"
            id={undefined}
            role="tooltip"
          >
            Sort
          </div>
        </div>
      </div>
    </th>
    <th
      aria-sort="ascending"
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-paddingCheckbox-13"
      scope="col"
    >
      <div
        className="MuiTooltip-root-15"
      >
        <span
          aria-describedby={undefined}
          className="MuiButtonBase-root-29 MuiTableSortLabel-root-24"
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          team
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-32 MuiTableSortLabel-icon-26 MuiTableSortLabel-iconDirectionDesc-27"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
        <div
          className="MuiTooltip-popper-16 MuiTooltip-popperClose-17"
          data-placement={undefined}
          data-x-out-of-boundaries={undefined}
          style={
            Object {
              "left": 0,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            aria-hidden={true}
            className="MuiTooltip-tooltip-18 MuiTooltip-tooltipPlacementBottom-22"
            id={undefined}
            role="tooltip"
          >
            Sort
          </div>
        </div>
      </div>
    </th>
    <th
      aria-sort="ascending"
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-paddingCheckbox-13"
      scope="col"
    >
      <div
        className="MuiTooltip-root-15"
      >
        <span
          aria-describedby={undefined}
          className="MuiButtonBase-root-29 MuiTableSortLabel-root-24"
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          points
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-32 MuiTableSortLabel-icon-26 MuiTableSortLabel-iconDirectionDesc-27"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
        <div
          className="MuiTooltip-popper-16 MuiTooltip-popperClose-17"
          data-placement={undefined}
          data-x-out-of-boundaries={undefined}
          style={
            Object {
              "left": 0,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            aria-hidden={true}
            className="MuiTooltip-tooltip-18 MuiTooltip-tooltipPlacementBottom-22"
            id={undefined}
            role="tooltip"
          >
            Sort
          </div>
        </div>
      </div>
    </th>
  </tr>
</thead>
`;

exports[`TableHeadContainer should render correctly with different props 1`] = `
<thead
  className="MuiTableHead-root-1"
>
  <tr
    className="MuiTableRow-root-2 MuiTableRow-head-3"
  >
    <th
      aria-sort="ascending"
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-paddingCheckbox-13"
      scope="col"
    >
      <div
        className="MuiTooltip-root-15"
      >
        <span
          aria-describedby={undefined}
          className="MuiButtonBase-root-29 MuiTableSortLabel-root-24"
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          position
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-32 MuiTableSortLabel-icon-26 MuiTableSortLabel-iconDirectionDesc-27"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
        <div
          className="MuiTooltip-popper-16 MuiTooltip-popperClose-17"
          data-placement={undefined}
          data-x-out-of-boundaries={undefined}
          style={
            Object {
              "left": 0,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            aria-hidden={true}
            className="MuiTooltip-tooltip-18 MuiTooltip-tooltipPlacementBottom-22"
            id={undefined}
            role="tooltip"
          >
            Sort
          </div>
        </div>
      </div>
    </th>
    <th
      aria-sort="ascending"
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-paddingCheckbox-13"
      scope="col"
    >
      <div
        className="MuiTooltip-root-15"
      >
        <span
          aria-describedby={undefined}
          className="MuiButtonBase-root-29 MuiTableSortLabel-root-24 MuiTableSortLabel-active-25"
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          team
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-32 MuiTableSortLabel-icon-26 MuiTableSortLabel-iconDirectionDesc-27"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
        <div
          className="MuiTooltip-popper-16 MuiTooltip-popperClose-17"
          data-placement={undefined}
          data-x-out-of-boundaries={undefined}
          style={
            Object {
              "left": 0,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            aria-hidden={true}
            className="MuiTooltip-tooltip-18 MuiTooltip-tooltipPlacementBottom-22"
            id={undefined}
            role="tooltip"
          >
            Sort
          </div>
        </div>
      </div>
    </th>
    <th
      aria-sort="ascending"
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-paddingCheckbox-13"
      scope="col"
    >
      <div
        className="MuiTooltip-root-15"
      >
        <span
          aria-describedby={undefined}
          className="MuiButtonBase-root-29 MuiTableSortLabel-root-24"
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          points
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-32 MuiTableSortLabel-icon-26 MuiTableSortLabel-iconDirectionDesc-27"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
        <div
          className="MuiTooltip-popper-16 MuiTooltip-popperClose-17"
          data-placement={undefined}
          data-x-out-of-boundaries={undefined}
          style={
            Object {
              "left": 0,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            aria-hidden={true}
            className="MuiTooltip-tooltip-18 MuiTooltip-tooltipPlacementBottom-22"
            id={undefined}
            role="tooltip"
          >
            Sort
          </div>
        </div>
      </div>
    </th>
  </tr>
</thead>
`;
